### ISSUE TITLE
iegraph: add option for bar label

### ISSUE THREAD

**Issue by luizaandrade** [worldbank/ietoolkit#142]
I got an email asking how to add the value of the bars as labels in iegraph. I could find out how to do it using the twoway graph options.

**Comment by kbjarkefur** [comment]
Did you forward the email? I saw your slack message. I think the solution might be in adding specifying that as a twoway option to the command as it is. There might be something about iegraph that makes it not work. I will look into how bar labels are added to stata graphs in general. I have I faint memory of already have done this, or helped someone do this using iegraph

**Comment by luizaandrade** [comment]
I couldn't find an easy way to add it with the twoway option, though it's pretty straight forward with bar graphs. I think the easiest thing is to add a scatter plot as mentioned [here](https://www.stata.com/statalist/archive/2007-01/msg00243.html):

**Comment by kbjarkefur** [comment]
Great link. That seems like a solution. And also something that requires an update to iegraph. I think we should do it.

**Comment by kbjarkefur** [comment]
I self assigned as I know the command well,  but I'd be perfectly fine with anyone else doing it

**Comment by kbjarkefur** [comment]
Oh, I saw that you wrote you had already implemented it. That sentence was cut out from my email notification

**Comment by luizaandrade** [comment]
Adding @alejandro-ome this thread -- he made the suggestions to add this feature.

**Comment by bbdaniels** [comment]
Last time I did this I allowed the option `barlabel()` to specify the relevant statistic at which to place the label. In this code it could instead be set by default to be placed at the `conf_int_max` of the the CI instead of the `mean`?

**Comment by kbjarkefur** [comment]
Its a good suggestion, but what if the CI is wide? Wont it look like the label shows the value of `conf_int_max` instead of the value of the bar?

**Comment by bbdaniels** [comment]
The other possibility I explored with these was just placing them at zero (inside the bar base). In the end I allowed the following:

```
[BARlab(string asis)]
  // Labels bars with means. Specify location: upper, lower, mean, zero.
...
local scatter x \`barlab' // Vertical bars
local scatter \`barlab' x // Horizontal bars
...
if "`barlab'" != "" {
  local blabplot ///
  (scatter `scatter', mlabel(mean2) msymbol(none) mlabc(black) mlabs(1.8) mlabp(3))
}
```

**Comment by luizaandrade** [comment]
Ooh, I like this last one! I think we should have both this and the clock position.

**Comment by bbdaniels** [comment]
zero is implemented just by creating a variable called `zero` that is equal to zero – in all these cases the available choices for the option match variable names that are in the summary dataset so they are slotted in just as is.

**Comment by kbjarkefur** [comment]
@luizaandrade and I think special cases like the one she mention in her comment right before this makes implementing set options like upper, mean etc. too difficult in relation to the benefit. So we will go with the solution in PR #143 but add that `mlabpos(12) mlabcolor(black)` is just the default and we will make it possible for user to edit the value in those two options.

**Comment by kbjarkefur** [comment]
Merged to develop branch in #143. This feature is added but addition is not yet published on SSC. This issue will be closed once it is.
